src/BlockID.cpp

```diff
@@ -222,6 +236,11 @@ bool StringToItem(const AString & a_ItemTypeString, cItem & a_Item)
 	{
 		ItemName = ItemName.substr(10);
 	}
+
+	if (gsBlockIDMap.m_bHasRunInit != true)
```

Comment:
Actually, never ever write `if (anything == true)` in C++, it may result in big surprises. `true` is a fixed constant, while the compiler is free to evaluate any true condition to **any** nonzero value, which will then compare as **not** equal to `true`.